byway/src/main/java/com/google/sps/servlets/PlacesServlet.java

```diff
@@ -52,4 +60,39 @@ private String formatLocation(PlaceType location) {
     place = place.replace('_', ' ');
     return place;
   }
+
+  /**
+   * Temporary setup to find entity with hard-coded id value. Set their interests.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query person =
+      new Query("user")
+        .setFilter(new FilterPredicate("id", FilterOperator.EQUAL, 2452));
+    PreparedQuery pq = datastore.prepare(person);
+    Entity userEntity = pq.asSingleEntity();
+    if(userEntity != null) {
+      String interestsAsString = request.getParameter("data");
+      addInterestsForUser(interestsAsString, userEntity, datastore);
+    } else {
+      response.setStatus(response.SC_NOT_FOUND);
+    }
+  }
+
+
+  /**
+   * Creates a list of interests from the JSON string passed in and adds it
+   * to this user's Entity object as a separate property.
+   * @param interestsAsString JSON String containing interests.
+   * @param currentUser       Entity with user information.
+   * @param datastore         DatastoreService with all users.
+   */
+  private void addInterestsForUser(String interestsAsString, Entity currentUser,
```

Comment:
if you declare datastore before your doPost you wont need to pass it as a param

Reply:
Fixed